soc/arm/silabs_exx32/Kconfig

```diff
@@ -26,6 +26,30 @@ config SOC_PART_NUMBER
 	  that you should not set directly. The part number selection choice defines
 	  the default value for this string.
 
+choice SOC_DCDC_GECKO_MODE
+	prompt "DC/DC mode"
+	depends on SOC_DCDC_GECKO
+	help
+	  Select power configuration mode of the on chip DC/DC converter.
+
+	config SOC_DCDC_GECKO_MODE_UNCONFIGURED
+		bool "Initial / Uconfigured"
+
+	config SOC_DCDC_GECKO_MODE_ON
+		bool "DC/DC On"
+
+	config SOC_DCDC_GECKO_MODE_OFF
+		bool "DC/DC Off"
+
+	config SOC_DCDC_GECKO_MODE_BYPASS
+		bool "Bypass"
+endchoice
+
+config SOC_DCDC_GECKO
+	bool "Enable SOC DCDC regulator"
+	help
+      Enable the DCDC for the SoC.
+
 config HAS_EMU
```

Comment:
Let's change the name of this option to `SOC_GECKO_EMU`. It should be enabled when used, as it's the case with DCDC converter, not merely when EMU module is present in the SoC. The option will also be enabled by the power management code which we should add soon.

Reply:
Do it as a separate commit.